Cython/Utility/CConvert.pyx

```diff
@@ -132,3 +132,79 @@ cdef inline tuple {{to_tuple_cname}}({{base_type}} *v, Py_ssize_t length):
         Py_INCREF(value)
         PyTuple_SET_ITEM(t, i, value)
     return t
+
+#################### c_array_slice.to_py ####################
+
+cdef extern from *:
+    void Py_INCREF(object o)
+    void Py_DECREF(object o)
+    object PyErr_Format(exc, const char *format, ...)
+
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object p, Py_ssize_t pos, object o)
+
+    tuple PyTuple_New(Py_ssize_t size)
+    void PyTuple_SET_ITEM(object p, Py_ssize_t pos, object o)
+
+    frozenset __Pyx_PyFrozenSet_New(object)
+    set PySet_New(void *)
+    int PySet_Add(object p, object o) except -1
+
+@cname("{{cname}}")
+cdef inline {{to_py_obj_type}} {{cname}}({{base_type}} *arr, Py_ssize_t arr_size, Py_ssize_t from_idx, Py_ssize_t to_idx, int wraparound, int boundscheck):
+    cdef Py_ssize_t i, length = 0
+
+    if wraparound:
+        if from_idx < 0: 
+            from_idx += arr_size
+
+        if to_idx < 0: 
+            to_idx += arr_size
+
+    if boundscheck:
+        if from_idx > to_idx:
+             from_idx = to_idx = 0
+        else:
+            if from_idx < 0:
+                from_idx = 0
+
+            if to_idx > arr_size:
+                to_idx = arr_size
+
+    length = to_idx - from_idx
+
+    {{if to_py_obj_type == 'list'}}
+    l = PyList_New(length)
```

Comment:
Could you rename this to `container`? `l` tends to look a lot like `1` or `I` in some fonts and is generally difficult to spot as a single character variable name, which makes the code less easy to read.
( `i` as an integer loop index variable is good since it plays in a different league.)

Reply:
ok